modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/helper/WikiSocialActivityHelper.java

```diff
@@ -48,6 +56,51 @@ public WikiSocialActivityHelper(WikiRequestHelper wikiRequestHelper) {
 		_wikiRequestHelper = wikiRequestHelper;
 	}
 
+	public List<SocialActivity> getApprovedSocialActivities(
+			WikiPage wikiPage, int start, int end)
+		throws PortalException {
+
+		WikiPage latestWikiPage = WikiPageLocalServiceUtil.getLatestPage(
+			wikiPage.getResourcePrimKey(), WorkflowConstants.STATUS_ANY, false);
+
+		if (latestWikiPage.getPageId() == wikiPage.getPageId()) {
+			return SocialActivityLocalServiceUtil.getActivities(
+				0, WikiPage.class.getName(), wikiPage.getResourcePrimKey(),
+				start, end);
+		}
+
+		DynamicQuery dynamicQuery =
+			SocialActivityLocalServiceUtil.dynamicQuery();
+
+		Property classPKProperty = PropertyFactoryUtil.forName("classPK");
+
+		dynamicQuery.add(classPKProperty.eq(wikiPage.getResourcePrimKey()));
+
+		Property extraDataProperty = PropertyFactoryUtil.forName("extraData");
+
+		Property typeProperty = PropertyFactoryUtil.forName("type");
+
+		dynamicQuery.add(
+			RestrictionsFactoryUtil.or(
+				RestrictionsFactoryUtil.not(
+					extraDataProperty.like(
+						"%version\":" + wikiPage.getVersion() + ",%")),
+				RestrictionsFactoryUtil.not(
+					typeProperty.in(
+						new int[] {
+							SocialActivityConstants.TYPE_ADD_ATTACHMENT,
+							SocialActivityConstants.
+								TYPE_MOVE_ATTACHMENT_TO_TRASH,
+							SocialActivityConstants.
+								TYPE_RESTORE_ATTACHMENT_FROM_TRASH
+						}))));
+
+		dynamicQuery.addOrder(OrderFactoryUtil.desc("createDate"));
+
+		return SocialActivityLocalServiceUtil.dynamicQuery(
+			dynamicQuery, start, end);
```

Comment:
Can you use the DSL query instead of a dynamic query?
```
		SocialActivityLocalServiceUtil.dslQuery(
			DSLQueryFactoryUtil.select(
				SocialActivityTable.INSTANCE
			).from(
				SocialActivityTable.INSTANCE
			).where(
				SocialActivityTable.INSTANCE.type.in(
					...
				).or(
					...
				)));
```
Unless you need to perform an action on the query results (where dynQuery is the way to go), dslQuery is preferred as it is generally more readable.

Reply:
Thanks @adolfopa, I've made the changes and force-pushed.